server/services/arduino/lib/device.setValue.js

```diff
@@ -0,0 +1,102 @@
+const { DEVICE_FEATURE_CATEGORIES, DEVICE_FEATURE_TYPES, DEVICE_FUNCTION, STATE } = require('../../../utils/constants');
+
+const logger = require('../../../utils/logger');
+
+/**
+ * @description Change value of a device.
+ * @param {Object} device - The device to control.
+ * @param {Object} deviceFeature - The binary deviceFeature to control.
+ * @param {string|number} value - The new value.
+ * @returns {Promise} - .
+ * @example
+ * setValue(device, deviceFeature, value);
+ */
+async function setValue(device, deviceFeature, value) {
+  const arduino = await this.gladys.device.getBySelector(
+    device.params.find((param) => param.name === 'ARDUINO_LINKED').value,
+  );
+  const path = arduino.params.find((param) => param.name === 'ARDUINO_PATH').value;
+
+  const functionName = device.params.find((param) => param.name === 'FUNCTION').value;
+
+  const message = {
+    function_name: functionName,
+    parameters: {
+      data_pin: device.params.find((param) => param.name === 'DATA_PIN').value,
+    },
+  };
+
+  switch (functionName) {
+    case DEVICE_FUNCTION.RECV_433:
+      // message.parameters['enable'] = value === 1 ? true : false;
+      if (
+        (deviceFeature.category === DEVICE_FEATURE_CATEGORIES.OPENING_SENSOR ||
+          deviceFeature.category === DEVICE_FEATURE_CATEGORIES.MOTION_SENSOR) &&
+        parseInt(device.params.find((param) => param.name === 'CODE').value, 0) === parseInt(value, 0)
```

Comment:
I would recommend that you use the helper function "getDeviceParam" here => https://github.com/GladysAssistant/Gladys/blob/master/server/utils/device.js#L12

Reply:
Thanks for the advice. I changed my scripts to replace every line by the "getDeviceParam" function 🙂